src/auth/src/token.rs

```diff
@@ -71,12 +72,12 @@ impl std::fmt::Debug for Token {
 
 #[async_trait::async_trait]
 pub(crate) trait TokenProvider: std::fmt::Debug + Send + Sync {
-    async fn token(&self) -> Result<Token>;
+    async fn token(&self) -> Result<Arc<Token>>;
```

Comment:
Do we need to change this?  Seems like any implementation of this method would be called once an hour or so as the `CachedTokenProvider` ... well, caches the value?